src/app/account-app/cart.ts

```diff
@@ -0,0 +1,59 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2019 Runbox Solutions AS (runbox.com).
+//
+// This file is part of Runbox 7.
+//
+// Runbox 7 is free software: You can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+
+import { ProductOrder } from './product-order';
+
+export class Cart {
+    items: ProductOrder[] = [];
+
+    constructor() {
+        const storedCart = localStorage.getItem('shoppingCart');
```

Comment:
would it be an idea to prepend / append user account id? if using multiple accounts in the same browser.

Reply:
Seems to be a recurring theme too: I had the same thing (and you had the same comment :)) in the offline calendar PR. Makes me think that perhaps we should have some common Service that handles the offline stuff, so it's harder to forget to do that?

Reply:
It required a bit of a redo of the cart and everything that uses it (ie the entire account app), but it's up now: https://github.com/runbox/runbox7/commit/0a71f7783c7fc3f1aea729574fb4d8b49c24b736.

It touches a lot of things and hopefully doesn't break too many of them. There will be tests for it too, but I'll appreciate a good luck at the async stuff, it was a bit of a PITA to get right.